### PR TITLE
feat: Add exclusion list support and improve user offboarding job

### PR DESCRIPTION
**Pull request by sunando94**
# Changes

- Move OffboardUserExclusionList type to types.go
- Add offboardUserExclusionListConfigName to AppConfig
- Update getUserListFromCache to handle empty keys gracefully
- Add logJobSummary function to consolidate logging
- Add test coverage for edge cases

## 📝 Description
- Add configurable exclusion list for user offboarding
- Rename exception list to exclusion list throughout codebase
- Optimize exclusion list lookup using map instead of slice (O(1) vs O(n))
- Add validation for empty userKeys with proper error handling
- Improve LDAP error handling for No Such Object errors
- Consolidate job summary logging into single function
- Add comprehensive tests for empty userKey and exclusion list scenarios
- Add exclusion list config name to app config for flexibility

### What changed?

Added safety checks

### Why is this change needed?

random users were getting dropped every 2 hours

### Dependencies

- N/A

---

### Test Coverage

Added additional test to test out the feature.

### Performance Impact

- N/A

---

### Deploy Steps

1. Create a yaml for exclusion list offboardUserExclusionListConfigName.

### Prerequisites

*Positive improvements*:
Exclusion list lookup optimization: Changed from O(n) slice iteration to O(1) map lookup
Before: For each user processed, the code iterated through the entire exclusion list (worst case: 100 comparisons for 100 excluded users)
After: Direct map lookup regardless of exclusion list size
Impact: For 1,000 users with 100 exclusions, reduced from ~100,000 comparisons to 1,000 map lookups
Measured: Theoretical complexity improvement; in practice, eliminates linear search overhead"

Early validation of empty userKeys: Added upfront check to skip empty/invalid userKeys before any processing
Impact: Prevents unnecessary LDAP queries and backend operations for invalid entries
Measured: Reduces processing time for invalid entries from full pipeline to single string check
Neutral/minor changes:
Consolidated logging into a single function has minimal performance impact (reduces function call overhead slightly)
Added config loading for exclusion list name (one-time cost during initialization)

### Post-Deployment Monitoring

- 

### Rollback Plan

- Deploy the previous branch.
- Update the config CR

---

## ⚠️ Breaking Changes

- [ ] This PR contains breaking changes
- [ ] Migration guide provided (if applicable)

**Details:**

- N/A

---

## ⚙️ Configuration Changes

- add offboardUserExclusionListConfigName 

---

## ✅ Developer Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added positive and negative tests that prove my fix is effective or that my feature works
- [ ] Relevant documentation (README, tech specs, etc.) has been added or updated
- [ ] All CI/CD checks are passing

### CODE REVIEW

File: pkg/config/loader.go [hunk 1]
```diff
@@ -180,3 +180,21 @@ func (c *Config) loadByConfigName(configName string, config interface{}) error {
...
+	cfg.viper.AddConfigPath(cfg.opts.configPath)
+	if err := cfg.viper.ReadInConfig(); err != nil {
+		return err
+	}
+	if err := cfg.viper.Unmarshal(config); err != nil {
+		return err
+	}
+	SubstituteConfigValues(reflect.ValueOf(config))
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (sunando94) [reply pkg/config/loader.go]: SubstituteConfigValues is not needed so removed.

File: internal/controller/periodicjobs/job_usernaut_offboarding.go [hunk 2]
```diff
@@ -437,13 +525,9 @@ func (uoj *UserOffboardingJob) isUserActiveInLDAP(ctx context.Context, userEmail
 	if err != nil {
 		if err == ldap.ErrNoUserFound {
 			// User not found in LDAP means they're inactive
+			uoj.logger.WithField("userEmail", userEmail).Debug("User not found in LDAP, treating as inactive")
 			return false, nil
 		}
```
[the remark this answers is not in the document]
Comment (sunando94) [reply internal/controller/periodicjobs/job_usernaut_offboarding.go]: Done and have pushed the change

File: appconfig/default.yaml [hunk 3]
```diff
@@ -114,4 +114,4 @@ apiServer:
     allowed_origins: ["http://localhost:3000", "http://localhost:8080"]
 
 usernautUserOffboardingInterval: "2h"
-
+offboardUserExclusionListConfigName: "default_offboard_user_exclusion_list"
```
Comment (vinamra28) [inline appconfig/default.yaml]:
```suggestion
offboardUserExclusionListConfigPath: "/path/to/exclusion_config.yaml"
```

File: internal/controller/periodicjobs/job_usernaut_offboarding.go [hunk 4]
```diff
@@ -101,12 +105,71 @@ func NewUserOffboardingJob(
...
+		// If file/URL doesn't exist or fails to load, log a warning and continue with empty exclusion list
+		log.WithFields(logrus.Fields{
+			"exclusionPath": configPath,
+		}).WithError(err).Warn("Exclusion list file/URL not found or failed to load, proceeding with empty exclusion list")
+		return
+	}
+
+	// Populate the exclusion list map with normalized email addresses for O(1) lookup
+	uoj.exclusionList = make(map[string]bool, len(exclusionList.Exclusions))
+	exclusionEmails := make([]string, 0, len(exclusionList.Exclusions))
```
Comment (vinamra28) [inline internal/controller/periodicjobs/job_usernaut_offboarding.go]: why do we need this array?

File: internal/controller/periodicjobs/job_usernaut_offboarding.go [hunk 5]
```diff
@@ -329,22 +413,28 @@ func (uoj *UserOffboardingJob) offboardUser(ctx context.Context, userKey string)
...
+	// Include list of offboarded users if any
+	fields := logrus.Fields{
+		"totalUsers":      totalUsers,
+		"offboardedUsers": result.offboardedCount,
+		"excludedCount":   result.excludedCount,
+		"errors":          len(result.errors),
+	}
+	if len(result.offboardedUsers) > 0 {
+		fields["removedUsers"] = result.offboardedUsers
 	}
```
Comment (vinamra28) [inline internal/controller/periodicjobs/job_usernaut_offboarding.go]: I guess it still makes to log the empty list
